### DISCUSSION TITLE
[Q]How to add styles for mergedcells

### DISCUSSION THREAD

**Discussion by fzpkhan** [exceljs/exceljs#2403] [Q&A]
Team,

Am looking for ways to add styling for merged cells. Is there way to do it? Like, I want my text to be center aligned in merged cells. Is there a way to apply styling for merged cells?

One more question i have is, Is there option to set property like colspan in a cell or row? If we support colspan property, it will reduce much programmatic work to merge cells?

**Comment by favio-effex** [comment]
is there any solutions ?

**Comment by skypesky** [comment]
@fzpkhan  @favio-effex 

- The following is just an example so you can try it out, if I remember correctly.

```typescript

// It is recommended to use typescript to get type hints

// create cell
const cell = worksheet.getCell(`1A:1E`);

// merge cell
worksheet.mergeCells(`1A:1E`);

// style
cell.fill = {
    type: 'pattern',
    pattern: 'solid',
    fgColor: {
      argb: 'FFADADAD',
    }
}
// style
cell.border =  {
    top: { style: 'medium', color: { argb: '00000000' } },
    left: { style: 'medium', color: { argb: '00000000' } },
    bottom: { style: 'medium', color: { argb: '00000000' } },
    right: { style: 'medium', color: { argb: '00000000' } }
}

// align
cell.alignment = {
	horizontal: 'center',
 },
```
